[Safe CPP] Address warnings in LayerOverlapMap

stwrt opened the pull request:
#### 9565391573f571677f96091cf5d3af6d2a7c88c4
<pre>
[Safe CPP] Address warnings in LayerOverlapMap
<a href="https://bugs.webkit.org/show_bug.cgi?id=302117">https://bugs.webkit.org/show_bug.cgi?id=302117</a>
<a href="https://rdar.apple.com/problem/164198178">rdar://problem/164198178</a>

Reviewed by Chris Dumez.

Address Safe CPP warnings LayerOverlapMap.cpp/h.

* Source/WebCore/SaferCPPExpectations/NoUncheckedPtrMemberCheckerExpectations
* Source/WebCore/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations
* Source/WebCore/rendering/LayerOverlapMap.cpp
* Source/WebCore/rendering/LayerOverlapMap.h
* Source/WebCore/rendering/RenderLayerCompositor.cpp

Canonical link: <a href="https://commits.webkit.org/302716@main">https://commits.webkit.org/302716@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e132dfa3af6c0389b5f42fa6a27cc46043f0b676

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/129926 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/2187 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/40784 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/137319 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/81419 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/e6e66f88-03cc-4528-9b8d-2ada749aa5a4) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/131797 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/2142 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/2078 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/98968 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/66778 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/18da65ab-3aeb-4f00-9fcb-425927d52b9a) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/132873 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/1606 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/116362 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/79661 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/6a22f88c-056d-4b46-9237-0094d42168be) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/1518 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/34489 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/80589 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/110028 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/34997 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/139800 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/1981 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/1833 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/107474 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/2026 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/112711 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/107362 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27339 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/1583 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/31181 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/54783 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/2054 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/65423 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/1869 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/1903 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/1977 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/rendering/LayerOverlapMap.cpp:
```diff
@@ -132,7 +132,7 @@ class OverlapMapContainer {
         }
 
-        const RenderLayer& layer;
+        CheckedRef<const RenderLayer> layer;
```
Can we mark the data member as `const`?

cdumez commented on Source/WebCore/rendering/LayerOverlapMap.cpp:
```diff
@@ -167,7 +167,7 @@ class OverlapMapContainer {
 
     ClippingScope m_rootScope;
-    const RenderLayer& m_scopeLayer;
+    CheckedRef<const RenderLayer> m_scopeLayer;
```
Can we mark the data member as `const`?

cdumez commented on Source/WebCore/rendering/LayerOverlapMap.h:
```diff
@@ -71,7 +71,7 @@ class LayerOverlapMap {
     Vector<std::unique_ptr<OverlapMapContainer>> m_speculativeOverlapStack;
     RenderGeometryMap m_geometryMap;
-    const RenderLayer& m_rootLayer;
+    CheckedRef<const RenderLayer> m_rootLayer;
```
Can we mark the data member as `const`?

cdumez reviewed (approved):
LGTM assuming the safer cpp bot is happy.

stwrt committed:
[Safe CPP] Address warnings in LayerOverlapMap
https://bugs.webkit.org/show_bug.cgi?id=302117
rdar://problem/164198178

Reviewed by Chris Dumez.

Address Safe CPP warnings LayerOverlapMap.cpp/h.

* Source/WebCore/SaferCPPExpectations/NoUncheckedPtrMemberCheckerExpectations
* Source/WebCore/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations
* Source/WebCore/rendering/LayerOverlapMap.cpp
* Source/WebCore/rendering/LayerOverlapMap.h
* Source/WebCore/rendering/RenderLayerCompositor.cpp

Canonical link: https://commits.webkit.org/302716@main
